discovery_engine_core/web-api/openapi.yaml

```diff
@@ -0,0 +1,189 @@
+openapi: 3.0.3
+info:
+  title: Personalization API
+  version: 1.0.0-rc1
+  description: |-
+    The endpoints can be contacted at `t<name>.xaynet.dev/v1`.
+    
+    # Auth
+    To authenticate with the API two bearer tokens will be provided. One will work for the
+    personalization endpoints and the other one will allow documents to be added to the system.
+    
+tags:
+  - name: personalization
+    description: Interaction of the user and personalized document
+  - name: ingestion
+    description: Add documents to the system
+paths:
+  /users/{user_id}/documents/personalized:
+    get:
+      tags:
+        - personalization
+      summary: Get personalized document for the user
+      description: |-
+        Returns a map from `document_id` to the score of that document for the
+        specified `user_id`. The score is a value between 0 and 1 where a higher value
+        means that the document matches the preferences of the user better.
+      parameters:
+        - name: user_id
+          in: path
+          description: Id of the user
+          required: true
+          schema:
+            $ref: '#/components/schemas/UserId'
+        - name: count
+          in: query
+          description: Maximum number of personalized documents to return
+          required: false
+          schema:
+            type: integer
+            format: int32
+            minimum: 1
+            maximum: 100
+            default: 10
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/PersonalizedDocumentsResponse'
+        '400':
+          description: invalid user id.
+        '404':
+          description: user not found.
+        '422':
+          description: impossible to create a personalized list for the user.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/PersonalizedDocumentsError'
+
+  /users/{user_id}/documents/interact:
```

Comment:
:question: Can't we not just keep with REST naming/structuring conventions and use nouns and put the document id in the path?

It's a update for an interaction with an user specific document. Which both are resources/entities of which there is more then one.

Reading the current path from a REST POV implies that either there is a single interaction for all documents or a document with the id `interact`. Given that document ids are client chosen there could be a random string id which happens to be `"interact"` which could become a problem with other hypothetical features which are user->document specific we might have in the future.

Reply:
As already said we can upload more the one document in the request.

We can have  `/users/{user_id}/interactions` instead.
And for the preferences we can have `/users/{user_id}/personalizedDocuments` as previously suggested by Simon.

I'm also ok with ditching the compliance with REST since it does not really cope well with non CRUD operations.

Reply:
Sorry I didn't saw the previous comment as it didn't show up in the diff view and was collapsed in the non-diff few.

Yes I guess the path you proposed here are nice.

REST can work nice if you just formulate things in the right way, but it doesn't work well at all with batching, like updating interactions for multiple documents at once.